plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -242,11 +245,36 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred opening
-	conn, err := sql.Open("mssql", server)
-	if err != nil {
-		return err
+	// deferred connection opening
+	var conn *sql.DB
+
+	if strings.Contains(strings.ToLower(server), strings.ToLower("password=")) {
+		// when password is provided in connection string, use SQL auth
+
```

Comment:
Can we log the database and server saying SQL Auth vs AAD auth? That would help in troubleshooting when something fails

Reply:
Yes, I think it'd be useful to have the Server/Database name and Auth mode details while troubleshooting. Once the above mentioned PR is merged, we can extend the metrics to capture this.